src/TxHistory/TxHistoryNavigator.tsx

```diff
@@ -0,0 +1,97 @@
+import {createStackNavigator} from '@react-navigation/stack'
+import React from 'react'
+import {StyleSheet, TouchableOpacity} from 'react-native'
+import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
+import {useSelector} from 'react-redux'
+
+import {Button} from '../../legacy/components/UiKit'
+import {UI_V2} from '../../legacy/config/config'
+import {defaultNavigationOptions, defaultStackNavigatorOptions} from '../../legacy/navigationOptions'
+import {TX_HISTORY_ROUTES, WALLET_ROOT_ROUTES} from '../../legacy/RoutesList'
+import {transactionsInfoSelector, walletMetaSelector} from '../../legacy/selectors'
+import {COLORS} from '../../legacy/styles/config'
+import {formatDateToSeconds} from '../../legacy/utils/format'
+import iconGear from '../assets/img/icon/gear.png'
+import {TxDetails} from './TxDetails'
+import {TxHistory} from './TxHistory'
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+const Stack = createStackNavigator<{
+  'tx-history-list': any
+  'tx-details': any
+}>()
+/* eslint-enable @typescript-eslint/no-explicit-any */
+
+export const TxHistoryNavigator = () => {
+  const walletMeta = useSelector(walletMetaSelector)
+  const transactionInfos = useSelector(transactionsInfoSelector)
+
+  return (
+    <Stack.Navigator screenOptions={defaultStackNavigatorOptions} initialRouteName={TX_HISTORY_ROUTES.MAIN}>
+      <Stack.Screen name={TX_HISTORY_ROUTES.MAIN} component={TxHistory} options={headerBarOptions(walletMeta.name)} />
+
+      <Stack.Screen
+        name={TX_HISTORY_ROUTES.TX_DETAIL}
+        component={TxDetails}
+        options={({route}) => ({
+          title: formatDateToSeconds(transactionInfos[route.params?.id].submittedAt),
+          ...defaultNavigationOptions,
+        })}
+      />
+    </Stack.Navigator>
+  )
+}
+
+const styles = StyleSheet.create({
+  settingsButton: {
+    height: '80%',
```

Comment:
in general, avoid percentages
also, i think in this case you can safely remove this style altogether